components/ws-daemon/pkg/controller/workspace_controller.go

```diff
@@ -302,7 +302,14 @@ func (wsc *WorkspaceController) handleWorkspaceStop(ctx context.Context, ws *wor
 		wsc.metrics.recordFinalizeTime(time.Since(disposeStart).Seconds(), ws)
 	}
 
-	wsc.emitEvent(ws, "Backup", disposeErr)
+	wsc.emitEvent(ws, "Backup", fmt.Errorf("failed to backup workspace: %w", disposeErr))
+
+	err = wsc.operations.DeleteWorkspace(ctx, ws.Name)
```

Comment:
how do we ensure that the workspace controller doesn't in the meantime remove the finalizer and delete the workspace resource (because the BackupComplete condition is added)? Is it OK if we don't complete the DeleteWorkspace call in rare occasions, e.g. if daemon restarts after Backup and before Delete, and when daemon comes back up the workspace resource is gone

Reply:
We have housekeeping in place that deletes abandoned workspaces, so even if the deletion of a workspace is interrupted it will be cleaned up after 5 minutes. It currently only takes care of mk1 workspaces but I will add one for mk2 as well. Just for getting started with testing mk2 in Dedicated the current approach is good enough in my opinion. We have to fix it for prod though.

Reply:
Ah ok, that works